src/coprocessor/codec/convert.rs

```diff
@@ -16,6 +16,65 @@ use std::borrow::Cow;
 
 use coprocessor::select::xeval::EvalContext;
 use super::Result;
+// `UNSPECIFIED_LENGTH` is unspecified length from FieldType
+pub const UNSPECIFIED_LENGTH: i32 = -1;
+
+pub fn truncate_str(mut s: String, flen: isize) -> String {
+    if flen != UNSPECIFIED_LENGTH as isize && s.len() > flen as usize {
+        s.truncate(flen as usize);
+        s
+    } else {
+        s
+    }
+}
+
+// `overflow` returns an overflowed error.
+#[macro_export]
+macro_rules! overflow {
+    ($val:ident, $bound:ident) => ({
+        Err(box_err!("constant {} overflows {}", $val, $bound))
+    });
+}
+
+// `convert_int_to_uint` converts an int value to an uint value.
+pub fn convert_int_to_uint(val: i64, upper_bound: u64, tp: u8) -> Result<u64> {
+    if val as u64 > upper_bound {
```

Comment:
Maybe because an u64 would be changed to i64 in `eval_int` while its `filed_type` is marked with unsigned. In some case we need to get the origin u64 just like this function did. @XuHuaiyu Do may words cover the meaning to you ?